File: admin-app/lib/api/contracts.dart
class ApiRequestContext {
  const ApiRequestContext({this.csrfToken});

  final String? csrfToken;
}

class CursorPage<T> {
  const CursorPage({required this.items, this.nextCursor});

  final List<T> items;
  final String? nextCursor;
}

class ApiFailure {
  const ApiFailure({
    required this.status,
    required this.code,
    required this.message,
    required this.requestId,
    this.details,
  });

  final int status;
  final String code;
  final String message;
  final String requestId;
  final Object? details;
}

class SseResumeCursor {
  const SseResumeCursor({this.lastEventId, this.after});

  final String? lastEventId;
  final int? after;
}

class SseMessage {
  const SseMessage({required this.id, required this.event, required this.data});

  final String id;
  final String event;
  final String data;
}

enum AgentVersionState { current, mismatch, unknown }

const supportedAgentVersion = '0.2.0';

class AgentStatusView {
  const AgentStatusView({
    required this.status,
    required this.versionState,
    this.name,
    this.version,
    this.hostname,
    this.architecture,
    this.lastSeenAt,
  });

  final String status;
  final AgentVersionState versionState;
  final String? name;
  final String? version;
  final String? hostname;
  final String? architecture;
  final String? lastSeenAt;
}

File: admin/src/api/contracts.ts
import type { ErrorResponse } from "./generated/models/ErrorResponse";

export interface ApiRequestContext {
  credentials: "include";
  csrfToken?: string;
}

export interface CursorPage<T> {
  items: T[];
  nextCursor?: string | null;
}

export interface ApiFailure {
  status: number;
  error: ErrorResponse;
}

export interface SseResumeCursor {
  lastEventId?: string;
  after?: number;
}

export interface SseMessage {
  id: string;
  event: string;
  data: string;
}

export type EnvEditorMode = "structured" | "raw";

export interface DotenvError {
  line: number;
  code: string;
  message: string;
}

export interface DotenvAssignment {
  kind: "assignment";
  key: string;
  value: string;
  quote: "'" | '"' | null;
}

export type DotenvLine = DotenvAssignment | { kind: "blank" | "comment" | "invalid"; raw: string };

export interface DotenvDocument {
  lines: DotenvLine[];
  errors: DotenvError[];
}
